jenkins-master/jobs/scripts/workspace/submission.py

```diff
@@ -0,0 +1,400 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import argparse
+import os
+import re
+import socket
+import time
+from urlparse import urljoin, urlparse
+import uuid
+
+from config import config
+
+here = os.path.dirname(os.path.abspath(__file__))
+
+LOOKUP_FRAGMENT = 'api/project/{repository}/resultset/?revision={revision}'
+REVISON_FRAGMENT = '/#/jobs?repo={repository}&revision={revision}'
+
+BUILD_STATES = ['running', 'completed']
+
+
+class JenkinsDefaultValueAction(argparse.Action):
+    """Jenkins cannot handle optional arguments to a CLI script so we pass in
+    a 'None' value by default.
+
+    """
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+        if nargs is not None:
+            raise ValueError("nargs not allowed")
+        argparse.Action.__init__(self, option_strings, dest, **kwargs)
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        if type(values) is str:
+            values = values if values != 'None' else None
+        elif type(values) is list:
+            values = [value if value != 'None' else None for value in values]
+
+        setattr(namespace, self.dest, values)
+
+
+class FirefoxUIResultParser(object):
```

Comment:
Please keep in mind that this is not the version we will use once we migrated to mozharness. I will update all that code similar to what Maja is using in her media tests. This is mostly a 1-1 copy beside the new `status` property which I already took from Maja's code.